darwin/future/core/types/query.py

```diff
@@ -1,10 +1,24 @@
 from __future__ import annotations
 
-from typing import List, Optional
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Callable, Dict, Generic, List, Optional, TypeVar
 
-from darwin.future.data_objects.team import Team
+from darwin.future.core.client import Client
 from darwin.future.pydantic_base import DefaultDarwin
 
+T = TypeVar("T", bound=DefaultDarwin)
+Param = Dict[str, Any]  # type: ignore
+
+
+class Modifiers(Enum):
+    GREATER_THAN = ">"
+    LESS_THAN = "<"
+    GREATER_EQUAL = ">="
+    LESS_EQUAL = "<="
+    NOT_EQUAL = "!="
+    CONTAINS = "contains"
```

Comment:
I'm assuming we don't need `NOT_CONTAINS` for now at least.  Do you think there are any others we do need for now.

Reply:
There could be others that we come up with but shouldn't be too hard to include when we do think of them.